python/ray/includes/object_ref.pxi

```
@@ -162,14 +163,13 @@ cdef class ClientObjectRef(ObjectRef):
             raise TypeError("Unexpected type for id {}".format(id))
 
     def __dealloc__(self):
-        if client is None or client.ray is None:
-            # Similar issue as mentioned in ObjectRef.__dealloc__ above. The
-            # client package or client.ray object might be set
-            # to None when the script exits. Should be safe to skip
-            # call_release in this case, since the client should have already
-            # disconnected at this point.
-            return
-        if client.ray.is_connected():
+        def _connected():
+            try:
+                return self._worker is not None and self._worker.is_connected()
+            except Exception:
```

Can we avoid swallowing exceptions?